src/Microsoft.AspNet.Mvc.Core/OptionDescriptors/ModelBinderDescriptorExtensions.cs

```diff
@@ -80,5 +80,24 @@ public static class ModelBinderDescriptorExtensions
             descriptors.Insert(index, descriptor);
             return descriptor;
         }
+
+
+        /// <summary>
+        /// Removes all instances of <typeparamref name="TInstance"/> from a list of ModelBinderDescriptors.
+        /// </summary>
+        /// <typeparam name="TInstance">A type that represents an <see cref="IModelBinder"/>.</typeparam>
+        /// <param name="descriptors">A list of ModelBinderDescriptors.</param>
+        public static void RemoveTypesOf<TInstance>(
+            [NotNull] this IList<ModelBinderDescriptor> descriptors)
+            where TInstance : class, IModelBinder
+        {
+            for (int i = descriptors.Count - 1; i >= 0; i--)
+            {
+                if (descriptors[i].OptionType == typeof(TInstance))
```

Comment:
No this should be specific

-----Original Message-----
From: "Doug Bunting" <notifications@github.com>
Sent: ‎2/‎7/‎2015 11:40 AM
To: "aspnet/Mvc" <Mvc@noreply.github.com>
Cc: "Yishai Galatzer" <yishai_galatzer@yahoo.com>
Subject: Re: [Mvc] [Fix for #1787] Add Extension Methods toIEnumerable<IOptionDescriptor<object>> (#1906)

In src/Microsoft.AspNet.Mvc.Core/OptionDescriptors/ModelBinderDescriptorExtensions.cs:
> @@ -80,5 +80,24 @@ public static class ModelBinderDescriptorExtensions
>              descriptors.Insert(index, descriptor);
>              return descriptor;
>          }
> +
> +
> +        /// <summary>
> +        /// Removes all instances of <typeparamref name="TInstance"/> from a list of ModelBinderDescriptors.
> +        /// </summary>
> +        /// <typeparam name="TInstance">A type that represents an <see cref="IModelBinder"/>.</typeparam>
> +        /// <param name="descriptors">A list of ModelBinderDescriptors.</param>
> +        public static void RemoveTypesOf<TInstance>(
> +            [NotNull] this IList<ModelBinderDescriptor> descriptors)
> +            where TInstance : class, IModelBinder
> +        {
> +            for (int i = descriptors.Count - 1; i >= 0; i--)
> +            {
> +                if (descriptors[i].OptionType == typeof(TInstance))
not positive of the expected behaviour here. should this use IsAssignableFrom()?
—
Reply to this email directly or view it on GitHub.